contracts/FlowIDTableStaking.cdc

```diff
@@ -874,6 +885,43 @@ pub contract FlowIDTableStaking {
             FlowIDTableStaking.account.save<{String: UFix64}>(nodeIDs, to: /storage/idTableNonOperationalNodesList)
         }
 
+        /// Removes nodes by setting their weight to zero and refunding
+        /// staked and delegated tokens.
+        access(self) fun removeAndRefundNodeRecord(nodeRecord: &NodeRecord) {
```

Comment:
Do we need to remove nodes from the candidate list in this code-path? 

For example, in `fillSlots`, we iterate over candidate nodes, and remove some of them using this function. Will those nodes be removed from the candidate list?

Reply:
The candidate node list is fully loaded and then reset anyway during `fillSlots`, so it doesn't matter for the execution of that function.

I think it might matter for when we're removing nodes during `removeUnapprovedNodes`. Nodes are already removed from the candidate node list if they unstake below the minimum, so that part of it isn't a problem, so the only edge-case that would require it would be if a candidate node is removed from the approved list in the same staking auction where they first staked, which is an extremely small edge-case, but still probably worth considering. I'll add a check for that in `removeAndRefundNodeRecord`